test/integration/volume/persistent_volumes_test.go

```diff
@@ -127,20 +127,20 @@ func TestPersistentVolumeRecycler(t *testing.T) {
 	// non-namespaced objects (PersistenceVolumes).
 	defer testClient.CoreV1().PersistentVolumes().DeleteCollection(context.TODO(), metav1.DeleteOptions{}, metav1.ListOptions{})
 
-	informers.Start(tCtx.Done())
-	go ctrl.Run(tCtx)
+	stopControllers := runControllerAndInformers(tCtx, ctrl, informers)
```

Comment:
Why we need this wrappr function? 

If `tCtx.Cancel("test has completed")` is called, the controller will also be stopped. We should not call `informers.WaitForCacheSync(ctx.Done())` manually in the integration test because this logic should be done by the controller itself. No `informers.WaitForCacheSync(ctx.Done())` may help us to find a missing `XXXSynced` in the `Run` function of the controller.

Reply:
revert. thanks